src/stream/src/executor_v2/top_n.rs

```diff
@@ -208,15 +208,6 @@ impl<S: StateStore> Executor for InnerTopNExecutor<S> {
     fn identity(&self) -> &str {
         &self.info.identity
     }
-
-    fn clear_cache(&mut self) -> Result<()> {
-        self.managed_lowest_state.clear_cache();
-        self.managed_middle_state.clear_cache();
-        self.managed_highest_state.clear_cache();
-        self.first_execution = true;
```

Comment:
This is quite suspicious. Why setting first_execution here before? Could that be related to @BugenZhao 's recently discovered bug?

By the way, we can remove no cache e2e test in CI.

Reply:
The HashJoin has not been migrated to executor v2 and its `clear_cache` still makes sense. Let's keep it for a while.

Reply:
IIRC, the `first_execution` indicates whether we need to fetch the state from storage and was necessary when writing this. 🤣 I'll investigate into TopN in next days. cc @lmatz